packages/clerk-js/src/ui/components/SessionTasks/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { useClerk } from '@clerk/shared/react';
+import { eventComponentMounted } from '@clerk/shared/telemetry';
+import type { SessionTask } from '@clerk/types';
+import { useCallback, useContext, useEffect } from 'react';
+
+import { SESSION_TASK_ROUTE_BY_KEY } from '../../../core/sessionTasks';
+import { OrganizationListContext, SignInContext, SignUpContext } from '../../../ui/contexts';
+import { Card, LoadingCardContainer, withCardStateProvider } from '../../../ui/elements';
+import {
+  SessionTasksContext as SessionTasksContext,
+  useSessionTasksContext,
+} from '../../contexts/components/SessionTasks';
+import { Route, Switch, useRouter } from '../../router';
+import { OrganizationList } from '../OrganizationList';
+
+const SessionTasksStart = withCardStateProvider(() => {
+  const clerk = useClerk();
+  const { navigate } = useRouter();
+  const { redirectUrlComplete } = useSessionTasksContext();
+
+  useEffect(() => {
+    // Simulates additional latency to avoid a abrupt UI transition when navigating to the next task
+    const timeoutId = setTimeout(() => {
+      void clerk.__experimental_nextTask({ redirectUrlComplete });
+    }, 500);
+    return () => clearTimeout(timeoutId);
```

Comment:
Just to make sure I'll parse correctly, that will display the loading spinner for 500ms and then redirect to the next task?

How do they land on this page? After a sign-in/sign-up (?)

Reply:
They land on this PR from a server-side redirection from `await auth()` -> it's implemented here https://github.com/clerk/javascript/pull/5440

Reply:
There's the argument of consistency, and always redirect to `tasks` on first `Clerk.nextTask` execution. 

I'm against that cause I think it's an unnecessary additional latency since we already have the client session loading. The whole point of this root path component is to resolve the client tasks since the server isn't aware of it.